### PR TITLE
3032 statefulset service name

### PR DESCRIPTION
**Pull request by mcminis1**
This PR addresses issue #3032 

ArgoCD expects statefulSets to have a serviceName. In this PR we add a service name to the two stateful sets that have this error.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3134)

### CODE REVIEW

File: kubeflow/gcp/iap.libsonnet [hunk 1]
```diff
@@ -276,6 +276,7 @@
           service: "backend-updater",
         },
       },
+      serviceName: "backend-updater",
```
Comment (lluunn) [inline kubeflow/gcp/iap.libsonnet]: we don't have such service, will it work?
Same below

Comment (mcminis1) [reply kubeflow/gcp/iap.libsonnet]: I have both of these in production. Seems to work fine. As far as I can tell this is the workaround.

It's an open (rotten) issue on the k8s repo.
https://github.com/kubernetes/kubernetes/issues/69608

**Comment by jlewi** [comment]
@mcminis1 Any idea why the tests are failing?

**Pull request merged**